source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,564 @@
+
+    { hello: 1, loadBalanced: 1 }
+
+**Data-Bearing Server Type**
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A :code:`ServerType` of :code:`LoadBalancer` MUST be considered a data-bearing server.
+
+
+**Driver Sessions**
+-------------------
+
+**How To Check Whether A Deployment Supports Sessions**
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Having verified in step 1 that the :code:`TopologyDescription` includes at least one
+connected server a driver can now determine whether sessions are supported by inspecting
+the :code:`TopologyType` and :code:`logicalSessionTimeoutMinutes` property. When the
+:code:`TopologyType` is :code:`LoadBalanced`, sessions are always supported.
+
+
+**Server Selection**
+--------------------
+
+Topology type: Load Balanced
+
+A deployment of topology type Load Balanced contains one server of type :code:`LoadBalancer`.
+
+For read and write operations, the single server in the topology MUST always be selected.
+
+For read operations the :code:`mode`, :code:`tag_sets`, and :code:`maxStalenessSeconds`
+MUST be passed through to the load balancer but do not affect selection.
+See `Passing read preference to mongos <https://bit.ly/2PbyV0B>`__.
+
+
+**Connection Pooling**
+----------------------
+
+When the driver is in load balancing mode and executing any cursor-initiating command, the driver
+MUST NOT check the connection back into the connection pool until the initial cursor
+or :code:`getMore` command has returned with an :code:`id` of :code:`0`. Otherwise the
+driver MUST continue to use the same connection for all :code:`getMore` and
+:code:`killCursors` commands for a given cursor. Drivers MUST check the connection back into
+pool on all :code:`cursor.close()` calls.
+
+For drivers that implement a connection pool and when using a pinned connection, the driver MUST
+emit only 1 :code:`ConnectionCheckOutStartedEvent`, and only 1 :code:`ConnectionCheckedOutEvent`
+or :code:`ConnectionCheckOutFailedEvent`. Similarly, the driver MUST only publish 1
+:code:`ConnectionCheckedInEvent`.
+
+For single threaded drivers that do not use a connection pool, the driver MUST have only 1
+socket connection to the load balancer in load balancing mode.
+
+In the case of network errors for all operations that create a cursor, the driver MUST behave
+as defined in the Retryable Reads specification. If a getMore fails with a network error,
+drivers MUST unpin the connection that’s pinned to the cursor and return it to the pool
+so it does not count against maxPoolSize. Future calls to close() on the cursor MUST NOT
+attempt to execute a killCursors command on the server.
+
+When the driver is in load balancing mode, drivers MUST NOT check the connection back into
+the connection pool while executing a transaction. The driver MUST continue to use the
+same connection to execute all commands for a given transaction with the exception
+of ::code:`commitTransaction` and :code:`abortTransaction`, if either one of these
```

Comment:
This wording implies that the rules in the paragraph below only apply to `commitTransaction` and `abortTransaction`, but the rule about unpinning on errors with the `TransientTransactionError` label actually applies to *all* commands in a transaction. We should probably also clarify that the rules for unpinning connections are the same as those for unpinning servers in non-LB sharded transactions and maybe link back to the transactions spec for those rules rather than duplicating them here.

Reply:
Linked back to transactions spec for these cases.